charts/internal/events/templates/_address.tpl

```diff
@@ -0,0 +1,12 @@
+{{- define "observe.collectorAddress" -}}
```

Comment:
We need an env var name we use everywhere. I was leaning towards `OBSERVE_COLLECT_URL`, which would imply we should call this `observe.collectUrl` here, but open to suggestions.

Reply:
I chose "address" rather than URL because this does not include the path. Since address is a more general term I think it's correct here, whereas URL would imply that they should give the full API path. In other words a URL identifies a resource and this address is just a component of that. Let me know what you think.

Reply:
Yea, this is always going to be a bit prone to confusion.  We have two components:
- the base URL for the API (set by the user)
- the path of the endpoint you intend to use (configured by the developer based on the data, `/v1/http`, `/v1/prometheus`, etc)

There's no reason to constrain the collector URL to expect just the domain name ("address"), and in practice your change doesn't really assume that (I think)? It's useful to be able to point at non-standard address for the purposes of local testing (e.g. "use this ngrok tunnel")

Reply:
I would avoid "address" except when referring to things explicitly described as addresses: IPs and MACs. Address is much less commonly used for hostnames. Once DNS and load balancers get involved the metaphor of an address as a unique locator for a specific machine is stretched thin. This is still a URL even if the path is extended in individual contexts. If you really want to call it a "base URL" you can but as long as the long form documentation is clear about this I don't think `base` is needed in variable names.

Reply:
Hmm I see address is a more generic term, such as "email address", or "TCP listen address" e.g. `0.0.0.0:8080` when configuring an HTTP server, etc. But OK, I see your point that really this is a URL with the path truncated.

Reply:
> observe.collectUrl

@jta I think the ship may have already sailed for `collect` vs `collector` in these values, as we unfortunately have customers already doing `observe.collectorHost` `observe.collectorPort` etc for these helm charts, so we'd break them if we changed to `collectHost` `collectPort` etc. And I think it would be an annoyance to people to have them inconsistent like `collectorPort` `collectorHost` `collectURL`.

Reply:
On the other hand, if we do decide that it's worth a break, I guess the sooner the better.

Reply:
Both of those are valid. URLs are addresses:

https://en.wikipedia.org/wiki/Address_space#Examples

Listen address is a good example of where address should be used, as there's no specific name for the `hostname:port` address format.

To be more practical, if you expect a protocol, I'd always call it a URL.

Reply:
@bendrucker counterpoint there is that socket addresses are defined as protocol + IP + port